src/AutoRest.CSharp/Common/Generation/Types/TypeFactory.cs

```diff
@@ -48,46 +48,61 @@ public TypeFactory(OutputLibrary library, Type unknownType)
+                InputPrimitiveTypeKind.AzureLocation => new CSharpType(typeof(AzureLocation), inputType.IsNullable),
+                InputPrimitiveTypeKind.Boolean => new CSharpType(typeof(bool), inputType.IsNullable),
+                InputPrimitiveTypeKind.Bytes => Configuration.ShouldTreatBase64AsBinaryData ? new CSharpType(typeof(BinaryData), inputType.IsNullable) : new CSharpType(typeof(byte[]), inputType.IsNullable),
+                InputPrimitiveTypeKind.ContentType => new CSharpType(typeof(ContentType), inputType.IsNullable),
+                InputPrimitiveTypeKind.PlainDate => new CSharpType(typeof(DateTimeOffset), inputType.IsNullable),
+                InputPrimitiveTypeKind.Decimal => new CSharpType(typeof(decimal), inputType.IsNullable),
+                InputPrimitiveTypeKind.Decimal128 => new CSharpType(typeof(decimal), inputType.IsNullable),
+                InputPrimitiveTypeKind.PlainTime => new CSharpType(typeof(TimeSpan), inputType.IsNullable),
+                InputPrimitiveTypeKind.ETag => new CSharpType(typeof(ETag), inputType.IsNullable),
+                InputPrimitiveTypeKind.Float32 => new CSharpType(typeof(float), inputType.IsNullable),
+                InputPrimitiveTypeKind.Float64 => new CSharpType(typeof(double), inputType.IsNullable),
+                InputPrimitiveTypeKind.Float128 => new CSharpType(typeof(decimal), inputType.IsNullable),
+                InputPrimitiveTypeKind.Guid or InputPrimitiveTypeKind.Uuid => new CSharpType(typeof(Guid), inputType.IsNullable),
+                InputPrimitiveTypeKind.Int8 => new CSharpType(typeof(sbyte), inputType.IsNullable),
+                InputPrimitiveTypeKind.UInt8 => new CSharpType(typeof(byte), inputType.IsNullable),
+                InputPrimitiveTypeKind.Int32 => new CSharpType(typeof(int), inputType.IsNullable),
+                InputPrimitiveTypeKind.Int64 => new CSharpType(typeof(long), inputType.IsNullable),
+                InputPrimitiveTypeKind.SafeInt => new CSharpType(typeof(long), inputType.IsNullable),
+                InputPrimitiveTypeKind.Integer => new CSharpType(typeof(long), inputType.IsNullable), // in typespec, integer is the base type of int related types, see type relation: https://typespec.io/docs/language-basics/type-relations
+                InputPrimitiveTypeKind.Float => new CSharpType(typeof(double), inputType.IsNullable), // in typespec, float is the base type of float32 and float64, see type relation: https://typespec.io/docs/language-basics/type-relations
+                InputPrimitiveTypeKind.Numeric => new CSharpType(typeof(double), inputType.IsNullable), // in typespec, numeric is the base type of number types, see type relation: https://typespec.io/docs/language-basics/type-relations
+                InputPrimitiveTypeKind.IPAddress => new CSharpType(typeof(IPAddress), inputType.IsNullable),
+                InputPrimitiveTypeKind.ArmId => new CSharpType(typeof(ResourceIdentifier), inputType.IsNullable),
```

Comment:
Looks like we lost arm-id in old version of TypeFactory... Anyway, it's better now.

Reply:
No, old version of TypeFactory has arm-id, its previous name is `ResourceIdentifier`. Because we are now using the kinds from TCGC, we have to rename it.